scripts/github_last_contribution.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python
+'''
+This script takes a list of Github users and delivers the timestamp of
+their last contribution.
+'''
+import argparse
+import json
+import requests
+
+from datetime import datetime, time
+
+GIT_API_ADDRESS = 'https://api.github.com/repos/{0}/{1}'
+
+def last_contribution(username, repo):
+    address = GIT_API_ADDRESS.format(username, repo)
+    request = requests.get(address)
+    if request.ok:
+        user_data = json.loads(request.text or request.content)
+        return user_data['pushed_at']
+    else:
+        print (("User {0}: Error while fetching data from Github "
+                "for the repository {1}! Status code: {2}")
+               .format(username, repo, request.status_code))
+
+def get_datetime(date):
+    return None if date is None else datetime.strptime(date, "%Y-%m-%dT%H:%M:%SZ")
+
+def date_output(repo, date):
+    if date is None:
+        print('This user has never contributed to zulip')
+    else:
+        print('Last {0} contribution: {1}'.format(repo, date))
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("usernames",
+                        help="Enter usernames separated with spaces.",
+                        type=str,
+                        nargs='+')
+    args = parser.parse_args()
+
+    result = []
+
+    for username in args.usernames:
+        date_zulip = get_datetime(last_contribution(username, 'zulip'))
+        date_zulip_gci = get_datetime(last_contribution(username, 'zulip-gci'))
+
+        if date_zulip is not None and date_zulip_gci is not None:
+            last_date = max(date_zulip, date_zulip_gci)
+        else:
+            last_date = date_zulip if date_zulip is not None else date_zulip_gci
+
+        result.append([last_date, date_zulip, date_zulip_gci, username])
+
+    result.sort(key=lambda entry: datetime.min if entry[0] is None else entry[0])
```

Comment:
That's right, but the problem is that the this first item entry[0] can sometimes be `None` (e.g. if the user was not found). In Python 3, any attempts at ordering NoneType instances result in an exception. This is why I need the built-in check.